### PR TITLE
2287 Enhance decollate to support list and Tensor

### LINKED ISSUE #2287: decollate needs to support more types
**Is your feature request related to a problem? Please describe.**
Currently, the `decollate_batch` API only supports `dict` data, need to support at least `list` type.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2287 .

### Description
This PR added `list` and `Tensor` support to the `decollate_batch` API, it can be useful to decollate the model output for post transforms and metrics.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by wyli — COMMENTED** [review]
thanks I put comments inline

File: monai/data/utils.py [hunk 1]
```diff
@@ -350,7 +351,26 @@ def decollate(data: Any, idx: int):
             return data[idx]
         raise TypeError(f"Not sure how to de-collate type: {type(data)}")
 
-    return [{key: decollate(data[key], idx) for key in data.keys()} for idx in range(batch_size)]
+    def _detect_batch_size(batch_data):
+        for v in batch_data:
+            if isinstance(v, torch.Tensor):
+                return v.shape[0]
+        raise RuntimeError("Couldn't determine batch size, please specify as argument.")
```
[the remark this answers is not in the document]
Comment (Nic-Ma) [reply monai/data/utils.py:358]: updated in the latest commit.
Thanks.

File: monai/data/utils.py [hunk 2]
```diff
@@ -350,7 +351,26 @@ def decollate(data: Any, idx: int):
...
+        batch_size = _detect_batch_size(batch_data=data.values()) if batch_size is None else batch_size
+        result = [{key: decollate(data[key], idx) for key in data.keys()} for idx in range(batch_size)]
+    elif isinstance(data, list):
+        batch_size = _detect_batch_size(batch_data=data) if batch_size is None else batch_size
+        result = [[decollate(d, idx) for d in data] for idx in range(batch_size)]
+    elif isinstance(data, torch.Tensor):
+        batch_size = data.shape[0]
+        result = [data[idx] for idx in range(batch_size)]
+    else:
+        raise RuntimeError("Only currently implemented for dictionary, list or Tensor data.")
```
[the remark this answers is not in the document]
Comment (Nic-Ma) [reply monai/data/utils.py]: updated in the latest commit.
Thanks.

**Review by wyli — APPROVED** [review]
thanks!

**Pull request merged**